File: 0601_0700/0691/0691.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    t = "ABCEHKMOPTXY"
    n = int(f.readline())
    for i in range(n):
        s = f.readline().strip()
        q.write("Yes\n" if len(s) == 6 and s[0] in t and s[4] in t and s[5] in t and s[1:4].isdigit() else "No\n")

File: 0601_0700/0691/0691.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n;
	char s[301];
	const char *t = "ABCEHKMOPTXY";

	fscanf(f, "%d", &n);
	while (n--)
	{
		fscanf(f, "%s", s);
		fprintf(q, strlen(s) == 6 && strchr(t, s[0]) && strchr(t, s[4]) && strchr(t, s[5]) && isdigit(s[1]) && isdigit(s[2]) && isdigit(s[3]) ? "Yes\n" : "No\n");
	}
	return 0;
}
